developer/src/kmc-ldml/src/compiler/tran.ts

```diff
@@ -216,17 +225,35 @@ export abstract class TransformCompiler<T extends TransformCompilerType, TranBas
     return defaults;
   }
 
-  /** true if ok */
-  private checkRanges(cookedFrom: string): string {
+  /** returns ===null on error */
```

Comment:
```suggestion
  /** @returns null if invalid escapes are found in `cookedFrom` string */
```

But what is an invalid escape? I can parse the function but still not clear exactly what it is we are checking

Reply:
it's invalid if it is in `\uXXXX` form without curly braces. I added a function comment.